Angular pwa installations fails

npm install @angular/pwa --save-dev
nx g @angular/pwa:ng-add --project=<project-name>

"nx": "14.3.1",

```
>  NX  Generating @angular/pwa:ng-add

[NX] Angular devkit called `writeWorkspace`, this may have had unintended consequences in workspace.json
[NX] Double check workspace.json before proceeding
TypeError: Cannot convert undefined or null to object
    at NxScopedHost.writeWorkspaceConfiguration (E:\Study\Projects\rms\node_modules\nx\src\adapter\ngcli-adapter.js:320:9)
    at E:\Study\Projects\rms\node_modules\nx\src\adapter\ngcli-adapter.js:215:29
    at E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\operators\switchMap.js:17:35
    at OperatorSubscriber._this._next (E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\operators\OperatorSubscriber.js:28:21)
    at OperatorSubscriber.Subscriber.next (E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\Subscriber.js:72:18)
    at E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\operators\switchMap.js:17:190
    at OperatorSubscriber._this._next (E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\operators\OperatorSubscriber.js:28:21)
    at OperatorSubscriber.Subscriber.next (E:\Study\Projects\rms\node_modules\rxjs\dist\cjs\internal\Subscriber.js:72:18)
    at SafeSubscriber.__tryOrUnsub (E:\Study\Projects\rms\node_modules\@angular-devkit\core\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.next (E:\Study\Projects\rms\node_modules\@angular-devkit\core\node_modules\rxjs\internal\Subscriber.js:143:22)
Cannot convert undefined or null to object

```

I am using 

node v16.15.1
npm v8.11.0

Could it be a problem?

I ran below commands:

```
npx create-nx-workspace rms (of type app)
npm install -D @nrwl/angular
nx g @nrwl/angular:application rms-client
npm install @angular/pwa --save-dev
nx g @angular/pwa:ng-add --project=rms-client
```

```
E:\Study\Projects\temp>npx create-nx-workspace rms
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
Need to install the following packages:
  create-nx-workspace
Ok to proceed? (y) y
√ What to create in the new workspace · apps
√ Use Nx Cloud? (It's free and doesn't require registration.) · No

 >  NX   Nx is creating your v14.3.3 workspace.

   To make sure the command works reliably in all environments, and that the preset is applied correctly,
   Nx will run "npm install" several times. Please wait.

√ Installing dependencies with npm
√ Nx has successfully created the workspace.

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————


 >  NX   First time using Nx? Check out this interactive Nx tutorial.

   https://nx.dev/core-tutorial/01-create-blog


E:\Study\Projects\temp>cd rms

E:\Study\Projects\temp\rms>npm install -D @nrwl/angular
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

added 254 packages, and audited 656 packages in 2m

65 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

E:\Study\Projects\temp\rms>nx g @nrwl/angular:application rms-client

>  NX  Generating @nrwl/angular:application

√ Which stylesheet format would you like to use? · scss
√ Would you like to configure routing for this application? (y/N) · true
UPDATE nx.json
UPDATE package.json
CREATE jest.config.ts
CREATE jest.preset.js
UPDATE .vscode/extensions.json
UPDATE .gitignore
UPDATE workspace.json
CREATE apps/rms-client/.browserslistrc
CREATE apps/rms-client/tsconfig.app.json
CREATE apps/rms-client/tsconfig.spec.json
CREATE apps/rms-client/src/favicon.ico
CREATE apps/rms-client/src/index.html
CREATE apps/rms-client/src/main.ts
CREATE apps/rms-client/src/polyfills.ts
CREATE apps/rms-client/src/styles.scss
CREATE apps/rms-client/src/assets/.gitkeep
CREATE apps/rms-client/src/environments/environment.prod.ts
CREATE apps/rms-client/src/environments/environment.ts
CREATE apps/rms-client/src/app/app.module.ts
CREATE apps/rms-client/src/app/app.component.html
CREATE apps/rms-client/src/app/app.component.spec.ts
CREATE apps/rms-client/src/app/app.component.ts
CREATE apps/rms-client/src/app/app.component.scss
CREATE apps/rms-client/tsconfig.editor.json
CREATE apps/rms-client/tsconfig.json
CREATE apps/rms-client/src/app/nx-welcome.component.ts
CREATE .eslintrc.json
CREATE apps/rms-client/.eslintrc.json
CREATE apps/rms-client/jest.config.ts
CREATE apps/rms-client/src/test-setup.ts
CREATE apps/rms-client-e2e/cypress.json
CREATE apps/rms-client-e2e/src/fixtures/example.json
CREATE apps/rms-client-e2e/src/integration/app.spec.ts
CREATE apps/rms-client-e2e/src/support/app.po.ts
CREATE apps/rms-client-e2e/src/support/commands.ts
CREATE apps/rms-client-e2e/src/support/index.ts
CREATE apps/rms-client-e2e/tsconfig.json
CREATE apps/rms-client-e2e/project.json
CREATE apps/rms-client-e2e/.eslintrc.json
CREATE apps/rms-client/project.json
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated

> rms@0.0.0 postinstall
> ngcc --properties es2020 browser module main


added 651 packages, removed 4 packages, changed 3 packages, and audited 1303 packages in 9m

150 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

E:\Study\Projects\temp\rms>npm install @angular/pwa --save-dev
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: jasmine-marbles@0.8.4
npm WARN Found: rxjs@7.4.0
npm WARN node_modules/rxjs
npm WARN   peer rxjs@"^6.5.3 || ^7.4.0" from @angular/common@14.0.1
npm WARN   node_modules/@angular/common
npm WARN     peer @angular/common@"14.0.1" from @angular/forms@14.0.1
npm WARN     node_modules/@angular/forms
npm WARN     4 more (@angular/platform-browser, ...)
npm WARN   4 more (@angular/core, @angular/forms, @angular/router, the root project)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer rxjs@"^6.5.3" from jasmine-marbles@0.8.4
npm WARN node_modules/jasmine-marbles
npm WARN   jasmine-marbles@"~0.8.4" from @nrwl/angular@14.3.3
npm WARN   node_modules/@nrwl/angular
npm WARN
npm WARN Conflicting peer dependency: rxjs@6.6.7
npm WARN node_modules/rxjs
npm WARN   peer rxjs@"^6.5.3" from jasmine-marbles@0.8.4
npm WARN   node_modules/jasmine-marbles
npm WARN     jasmine-marbles@"~0.8.4" from @nrwl/angular@14.3.3
npm WARN     node_modules/@nrwl/angular
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: rxjs-for-await@0.0.2
npm WARN Found: rxjs@7.4.0
npm WARN node_modules/rxjs
npm WARN   peer rxjs@"^6.5.3 || ^7.4.0" from @angular/common@14.0.1
npm WARN   node_modules/@angular/common
npm WARN     peer @angular/common@"14.0.1" from @angular/forms@14.0.1
npm WARN     node_modules/@angular/forms
npm WARN     4 more (@angular/platform-browser, ...)
npm WARN   4 more (@angular/core, @angular/forms, @angular/router, the root project)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer rxjs@"^6.0.0" from rxjs-for-await@0.0.2
npm WARN node_modules/rxjs-for-await
npm WARN   rxjs-for-await@"0.0.2" from @nrwl/angular@14.3.3
npm WARN   node_modules/@nrwl/angular
npm WARN
npm WARN Conflicting peer dependency: rxjs@6.6.7
npm WARN node_modules/rxjs
npm WARN   peer rxjs@"^6.0.0" from rxjs-for-await@0.0.2
npm WARN   node_modules/rxjs-for-await
npm WARN     rxjs-for-await@"0.0.2" from @nrwl/angular@14.3.3
npm WARN     node_modules/@nrwl/angular

added 13 packages, removed 15 packages, changed 2 packages, and audited 1302 packages in 23s

150 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

E:\Study\Projects\temp\rms>nx g @angular/pwa:ng-add --project rms-client

>  NX  Generating @angular/pwa:ng-add

[NX] Angular devkit called `writeWorkspace`, this may have had unintended consequences in workspace.json
[NX] Double check workspace.json before proceeding
TypeError: Cannot convert undefined or null to object
    at NxScopedHost.writeWorkspaceConfiguration (E:\Study\Projects\temp\rms\node_modules\nx\src\adapter\ngcli-adapter.js:320:9)
    at E:\Study\Projects\temp\rms\node_modules\nx\src\adapter\ngcli-adapter.js:215:29
    at E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\operators\switchMap.js:17:35
    at OperatorSubscriber._this._next (E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\operators\OperatorSubscriber.js:28:21)
    at OperatorSubscriber.Subscriber.next (E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\Subscriber.js:72:18)
    at E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\operators\switchMap.js:17:190
    at OperatorSubscriber._this._next (E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\operators\OperatorSubscriber.js:28:21)
    at OperatorSubscriber.Subscriber.next (E:\Study\Projects\temp\rms\node_modules\rxjs\dist\cjs\internal\Subscriber.js:72:18)
    at SafeSubscriber.__tryOrUnsub (E:\Study\Projects\temp\rms\node_modules\@angular-devkit\core\node_modules\rxjs\internal\Subscriber.js:205:16)
    at SafeSubscriber.next (E:\Study\Projects\temp\rms\node_modules\@angular-devkit\core\node_modules\rxjs\internal\Subscriber.js:143:22)
Cannot convert undefined or null to object

```

--- a/packages/nx/src/adapter/ngcli-adapter.ts
+++ b/packages/nx/src/adapter/ngcli-adapter.ts
@@ -436,7 +436,7 @@ export class NxScopedHost extends virtualFs.ScopedHost<any> {
         if (formatted) {
           const { cli, generators, defaultProject, ...workspaceJson } =
             formatted;
-          delete cli.schematicCollections;
+          delete cli?.schematicCollections;
           return merge(
             this.writeWorkspaceConfigFiles(context, workspaceJson),
             cli || generators || defaultProject
@@ -451,7 +451,7 @@ export class NxScopedHost extends virtualFs.ScopedHost<any> {
             defaultProject,
             ...angularJson
           } = w;
-          delete cli.schematicCollections;
+          delete cli?.schematicCollections;
           return merge(
             this.writeWorkspaceConfigFiles(context, angularJson),
             cli || schematics
@@ -467,7 +467,7 @@ export class NxScopedHost extends virtualFs.ScopedHost<any> {
     }
     const { cli, schematics, generators, defaultProject, ...angularJson } =
       config;
-    delete cli.schematicCollections;
+    delete cli?.schematicCollections;
 
     return merge(
       this.writeWorkspaceConfigFiles(context, angularJson),

